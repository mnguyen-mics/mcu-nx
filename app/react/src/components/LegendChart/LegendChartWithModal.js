import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal, Row, Col, Radio } from 'antd';
import McsIcon from '../McsIcon.tsx';

class LegendChartWithModal extends Component {

  constructor(props) {
    super(props);

    this.state = {
      key1: '',
      key2: '',
    };
  }

  render() {
    const { options } = this.props;
    const renderPicker = (that) => {
      const { legends } = this.props;
      const radioStyle = {
        display: 'block',
        height: '30px',
        lineHeight: '30px',
      };

      that.setState({
        key1: options[0].key,
        key2: options[1].key,
      });

      const onChangeValueLeft = e => {
        that.setState({ key1: e.target.value });
      };
      const onChangeValueRight = e => {
        that.setState({ key2: e.target.value });
      };

      return (
        <Row>
          <Col span={12}>
            <Radio.Group
              defaultValue={options[0].key}
              onChange={e => {
                onChangeValueLeft(e, 0);
              }}
            >
              {legends.map(legend => {
                return (
                  <Radio key={legend.key} style={radioStyle} value={legend.key}>
                    {legend.domain}
                  </Radio>
                );
              })}
            </Radio.Group>
          </Col>
          <Col span={12}>
            <Radio.Group
              defaultValue={options[1].key}
              onChange={e => onChangeValueRight(e, 1)}
            >
              {legends.map(legend => {
                return (
                  <Radio key={legend.key} style={radioStyle} value={legend.key}>
                    {legend.domain}
                  </Radio>
                );
              })}
            </Radio.Group>
          </Col>
        </Row>
      );
    };

    const renderModal = (that) => {
      const { onLegendChange } = this.props;
      Modal.confirm({
        title: 'Please select two items to display',
        content: renderPicker(that),
        onOk() {
          onLegendChange(that.state.key1, that.state.key2);
        },
        onCancel() {
        },
      });
    };

    return (
      <div className="mcs-legend-container">
        {options.map(option => {
          return (
            <div key={option.domain} style={{ float: 'left' }}>
              <div
                style={{
                  backgroundColor: option.color,
                  marginLeft: '10px',
                  width: '30px',
                  height: '4px',
                  borderRadius: '5px',
                  marginTop: '18px',
                  float: 'left',
                }}
              />
              <span
                style={{
                  float: 'right',
                  lineHeight: '40px',
                  marginLeft: '5px',
                }}
              >{option.domain}</span>
            </div>
          );
        })}
        <div
          style={{
            float: 'left',
            marginLeft: '10px',
            borderLeft: '1px solid #f0f0f0',
            height: '40px',
            lineHeight: '40px',
          }}
        >
          <button
            className="mcs-invisible-button"
            onClick={() => {
              renderModal(this);
            }}
          >
            <McsIcon
              style={{ marginLeft: '10px', color: '#d0d0d0' }}
              type="pen"
            />
          </button>
        </div>
      </div>
    );
  }
}

LegendChartWithModal.defaultProps = {
  legends: [],
  onLegendChange: () => {},
};

LegendChartWithModal.propTypes = {
  legends: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      domain: PropTypes.string.isRequired,
    }),
  ),
  onLegendChange: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      keys: PropTypes.arrayOf(PropTypes.string),
      colors: PropTypes.arrayOf(PropTypes.string),
      domains: PropTypes.arrayOf(PropTypes.string),
    }),
  ).isRequired,
};

export default LegendChartWithModal;
