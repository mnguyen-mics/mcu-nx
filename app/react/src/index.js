import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
