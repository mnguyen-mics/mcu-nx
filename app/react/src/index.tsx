/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
