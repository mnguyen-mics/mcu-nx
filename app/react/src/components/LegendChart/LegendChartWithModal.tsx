import * as React from 'react';
import { Modal, Row, Col, Radio } from 'antd';
import McsIcon from '../McsIcon';

interface Legend {
  key: string;
  domain: string;
}

interface LegendOption {
  key: string;
  color: string;
  domain: string;
}

interface LegendChartWithModalProps {
  identifier: string;
  options: LegendOption[];
  legends: Legend[];
  onLegendChange: (a: string, b: string) => void;
}

interface LegendChartWithModalState {
  key1: string;
  key2: string;
}

class LegendChartWithModal extends React.Component<LegendChartWithModalProps, LegendChartWithModalState> {

  constructor(props: LegendChartWithModalProps) {
    super(props);

    this.state = {
      key1: '',
      key2: '',
    };
  }

  render() {
    const { options } = this.props;
    const renderPicker = () => {
      const { legends } = this.props;
      const radioStyle = {
        display: 'block',
        height: '30px',
        lineHeight: '30px',
      };

      this.setState({
        key1: options[0].key,
        key2: options[1].key,
      });

      const onChangeValueLeft = (e: any) => {
        this.setState(
          (oldState: LegendChartWithModalState) => {
            return {
              ...oldState,
              key1: e.target.value,
            };
          });
      };
      const onChangeValueRight = (e: any) => {
        this.setState(
      (oldState: LegendChartWithModalState) => {
          return {
            ...oldState,
            key2: e.target.value,
          };
        });
      };

      return (
        <Row>
          <Col span={12}>
            <Radio.Group
              defaultValue={options[0].key}
              onChange={onChangeValueLeft}
            >
              {legends.map(legend => {
                return (
                  <Radio key={legend.key} style={radioStyle} value={legend.key}>
                    {legend.domain}
                  </Radio>
                );
              })}
            </Radio.Group>
          </Col>
          <Col span={12}>
            <Radio.Group
              defaultValue={options[1].key}
              onChange={onChangeValueRight}
            >
              {legends.map(legend => {
                return (
                  <Radio key={legend.key} style={radioStyle} value={legend.key}>
                    {legend.domain}
                  </Radio>
                );
              })}
            </Radio.Group>
          </Col>
        </Row>
      );
    };

    const renderModal = (that: LegendChartWithModal) => {
      const { onLegendChange } = this.props;
      Modal.confirm({
        title: 'Do you Want to delete these items?',
        content: renderPicker(),
        onOk() {
          onLegendChange(that.state.key1, that.state.key2);
        },
        onCancel() {
          return;
        },
      });
    };

    const onClick =  () => renderModal(this);
    return (
      <div className="mcs-legend-container">
        {options.map(option => {
          return (
            <div key={option.domain} style={{ float: 'left' }}>
              <div
                style={{
                  backgroundColor: option.color,
                  marginLeft: '10px',
                  width: '30px',
                  height: '4px',
                  borderRadius: '5px',
                  marginTop: '18px',
                  float: 'left',
                }}
              />
              <span
                style={{
                  float: 'right',
                  lineHeight: '40px',
                  marginLeft: '5px',
                }}
              >
                {option.domain}
              </span>
            </div>
          );
        })}
        <div
          style={{
            float: 'left',
            marginLeft: '10px',
            borderLeft: '1px solid #f0f0f0',
            height: '40px',
            lineHeight: '40px',
          }}
        >
          <button
            className="mcs-invisible-button"
            onClick={onClick}
          >
            <McsIcon
              style={{ marginLeft: '10px', color: '#d0d0d0' }}
              type="pen"
            />
          </button>
        </div>
      </div>
    );
  }
}

export default LegendChartWithModal;
