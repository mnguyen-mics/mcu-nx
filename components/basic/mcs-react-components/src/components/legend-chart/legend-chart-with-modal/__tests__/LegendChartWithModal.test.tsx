import 'jest';
import * as React from 'react';
import LegendChartWithModal, { LegendChartWithModalProps } from '../LegendChartWithModal';
import * as TestRenderer from 'react-test-renderer';

it('renders the legendchartWithModal', () => {
  const props: LegendChartWithModalProps = {
    options: [
      {
        key: '1',
        color: '#003056',
        domain: 'Domain #1',
      },
      {
        key: '2',
        color: '#00ab67',
        domain: 'Domain #2',
      },
    ],
    legends: [
      {
        key: '1',
        domain: 'Domain #1',
      },
      {
        key: '2',
        domain: 'Domain #2',
      },
    ],
    onLegendChange: (a, b) => {
      //
    },
  };

  const component = TestRenderer.create(<LegendChartWithModal {...props} />);

  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
