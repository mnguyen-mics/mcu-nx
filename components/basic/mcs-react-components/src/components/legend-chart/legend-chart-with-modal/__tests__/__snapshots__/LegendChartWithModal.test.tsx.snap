// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the legendchartWithModal 1`] = `
<div
  className="mcs-legend-container "
>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#003056",
          "borderRadius": "5px",
          "float": "left",
          "height": "4px",
          "marginLeft": "10px",
          "marginTop": "18px",
          "width": "30px",
        }
      }
    />
    <span
      style={
        Object {
          "float": "right",
          "lineHeight": "40px",
          "marginLeft": "5px",
        }
      }
    >
      Domain #1
    </span>
  </div>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#00ab67",
          "borderRadius": "5px",
          "float": "left",
          "height": "4px",
          "marginLeft": "10px",
          "marginTop": "18px",
          "width": "30px",
        }
      }
    />
    <span
      style={
        Object {
          "float": "right",
          "lineHeight": "40px",
          "marginLeft": "5px",
        }
      }
    >
      Domain #2
    </span>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid #f0f0f0",
        "float": "left",
        "height": "40px",
        "lineHeight": "40px",
        "marginLeft": "10px",
      }
    }
  >
    <button
      className="mcs-invisible-button"
      onClick={[Function]}
    >
      <span
        className="mcs-icon "
        style={
          Object {
            "color": "#d0d0d0",
            "marginLeft": "10px",
          }
        }
      >
        <i
          className="mcs-pen"
          style={Object {}}
        />
      </span>
    </button>
  </div>
</div>
`;
